core/src/main/java/org/commonjava/propulsor/boot/Booter.java

```diff
@@ -137,7 +137,7 @@ public static BootOptions loadBootOptionsByServiceLoader() throws BootException
 
     public static void setUncaughtExceptionHandler()
     {
-        Thread.currentThread().setUncaughtExceptionHandler( ( thread, error ) -> {
+        Thread.setDefaultUncaughtExceptionHandler( ( thread, error ) -> {
```

Comment:
Sorry for some misunderstanding. No need to change this here. Only need to extract that lambda.

Reply:
NP. I think this is needed. The currentThread.setXX is not useful. Eventually we need the setDefaultXXX.